src/chains/migrations/0031_chain_block_explorer_uri_api_template.py

```diff
@@ -0,0 +1,53 @@
+# Generated by Django 3.2.9 on 2021-11-03 09:54
+from django.apps.registry import Apps
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations import RunPython
+
+
+def set_api_url(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    Chain = apps.get_model("chains", "Chain")
+    for row in Chain.objects.all():
+        row.block_explorer_uri_api_template = API_TEMPLATE_MAPPING.get(row.id, "")
+        row.save(update_fields=["block_explorer_uri_api_template"])
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("chains", "0030_wallet"),
+    ]
+
+    operations = [
+        # Add the field block_explorer_uri_api_template with null=True in order to create a valid column
+        migrations.AddField(
+            model_name="chain",
+            name="block_explorer_uri_api_template",
+            field=models.URLField(null=True),
+            preserve_default=False,
+        ),
+        # Fill the column with the values from API_TEMPLATE_MAPPING (empty string if value is not found)
+        migrations.RunPython(set_api_url, RunPython.noop),
+        # set the block_explorer_uri_api_template column as non-nullable
+        migrations.AlterField(
+            model_name="chain",
+            name="block_explorer_uri_api_template",
+            field=models.URLField(null=False),
+            preserve_default=False,
+        ),
+    ]
+
+
+QUERY_PARAM_TEMPLATE = (
+    "?module={{module}}&action={{action}}&address={{address}}&apiKey={{apiKey}}"
+)
+
+API_TEMPLATE_MAPPING = {
+    1: f"https://api.etherscan.io/api{QUERY_PARAM_TEMPLATE}",
+    4: f"https://api-rinkeby.etherscan.io/api{QUERY_PARAM_TEMPLATE}",
+    56: f"https://api.bscscan.com/api{QUERY_PARAM_TEMPLATE}",
+    100: f"https://blockscout.com/poa/xdai/api{QUERY_PARAM_TEMPLATE}",
+    137: f"https://api.polygonscan.com/api{QUERY_PARAM_TEMPLATE}",
+    246: f"https://explorer.energyweb.org/api{QUERY_PARAM_TEMPLATE}",
+    42161: f"https://api.arbiscan.io/api{QUERY_PARAM_TEMPLATE}",
+    73799: f"https://volta-explorer.energyweb.org/api{QUERY_PARAM_TEMPLATE}",
```

Comment:
These chainIds can potentially be featured in other migrations, maybe it would make sense to use constants? Just a suggestion, I realize it's just a migration.